### ISSUE TITLE
E106 is raised if subdirectories are present in the "tasks" folder

### ISSUE THREAD

**Issue by nre-ableton** [ansible/ansible-lint#1001]
##### Summary

Sorry to report this bug after 4.3.2 was released; I didn't have time to test ansible-lint from `master` before the release was cut. Anyways, the fix made in https://github.com/ansible/ansible-lint/pull/980 does not work if a role has subdirectories in the `tasks` folder, for example:

```
ansible/roles/foo.bar/tasks/debian/install.yml:1: [E106] Role name foo.bar does not match ``^[a-z][a-z0-9_]+$`` pattern
```

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details
ansible-lint --version: 4.3.2

- ansible installation method: pip
- ansible-lint installation method: pip

**Comment by ssbarnea** [comment]
Ohh joy. Please make a PR and I can help making a new release. Include a test to avoid regression.

**Comment by nre-ableton** [comment]
@ssbarnea I'm not terribly familiar with the codebase, but I'll give it a shot!

**Comment by nre-ableton** [comment]
Ok, I finally got around to looking at this issue. Fix is here: https://github.com/ansible/ansible-lint/pull/1044
